File: Sources/APTradeDomain/NewsCategory.swift
/// A market-news category. Conceptual — the news-source query mapping lives in
/// infrastructure.
public enum NewsCategory: String, CaseIterable, Sendable {
    case general, crypto, merger

    public var displayName: String {
        switch self {
        case .general: return "General"
        case .crypto: return "Crypto"
        case .merger: return "Merger"
        }
    }
}

File: shared/src/commonMain/kotlin/com/aptrade/shared/domain/NewsCategory.kt
package com.aptrade.shared.domain

/**
 * A market-news category. Conceptual — the news-source query mapping lives in
 * infrastructure. Declaration order is the UI order.
 */
enum class NewsCategory(val finnhubValue: String, val displayName: String) {
    General("general", "General"),
    Crypto("crypto", "Crypto"),
    Merger("merger", "Merger"),
}
